Initial Trello powerup development

samuelhwilliams opened the pull request:
## Summary
The creation of a Power-up for Trello that adds a button to all cards allowing full(ish) history of activity on that card to be viewed. This makes it possible to see how knowledge and understanding has changed over time and also provides a more thorough audit trail of what's happened with the card.

## Caveats
Trello does not seem to make certain events available:
* Comment edits are not available; the original comment text is simply updated.
* Comments that are deleted are purged from the record entirely.
* Creation, modification and deletion of individual checklist items are not recorded. The creation and deletion of a checklist *is* visible, as well as marking checklist items complete/incomplete.

## Example
![trello-history](https://user-images.githubusercontent.com/2920760/27795842-8920ec44-5fff-11e7-8687-323b392b6615.gif)

Commit:
Initial Trello powerup development

galund commented on public/js/activity-builder.js:
```diff
@@ -0,0 +1,321 @@
+  attachment.classList.add('js-open-attachment-viewer');
+  attachment.classList.add('js-friendly-links');
+
+  if (typeof item.data.attachment.url === 'string') {
```
Are there other options as to what it might be??

galund commented on public/js/activity-builder.js:
```diff
@@ -0,0 +1,321 @@
+
+ActivityBuilder.prototype._addBoldTextNode = function _addBoldTextNode(parent, text) {
+  var child = document.createElement('SPAN');
+  child.classList.add('u-font-weight-bold');
```
ewww for classes that mention the styles they are (currently). You might as well just add a `<b>` rather than a SPAN :)

samuelhwilliams replied:
It's a hook into Trello's CSS, so using that to remain consistent with however they style it.

allait commented on manifest.yml:
```diff
@@ -0,0 +1,10 @@
+  memory: 64M
+  instances: 1
+  env:
+    TRELLO_DEVELOPER_KEY: 303537c2401a020448cd1f999fd6db29
```
Is it ok to have this here? Does it allow access to the power-up data?

samuelhwilliams commented:
@allait Have deleted the Trello key from here and put it in CF's environment manually. Updated the readme to note this. Deleted your comment because it displayed the key.

galund commented:
(We originally said that trello key was okay in the clear because it goes into the client environment anyway? I do think it should be in source control somewhere really, and encrypting it seems a faff?)

samuelhwilliams commented:
@allait preferred it out of here because it could be grabbed by anyone on GitHuby. Trello renders it into the client environment, but that's only visible to be people who've installed the Power-Up we're hosting - i.e. just GDS, in theory. I think that's really the only reason. It's not super secret, but we can make a cursory effort to limit exposure anyway.
